prow/git/v2/client_factory.go

```diff
@@ -245,22 +258,42 @@ func (c *clientFactory) bootstrapClients(org, repo, dir string) (cacher, cloner,
 	return client, client, client, nil
 }
 
-// ClientFromDir returns a repository client for a directory that's already initialized with content.
+// ClientFromDirHost returns a repository client for a directory that's already initialized with content.
 // If the directory isn't specified, the current working directory is used.
-func (c *clientFactory) ClientFromDir(org, repo, dir string) (RepoClient, error) {
-	_, _, client, err := c.bootstrapClients(org, repo, dir)
+func (c *clientFactory) ClientFromDirHost(host, org, repo, dir string) (RepoClient, error) {
+	_, _, client, err := c.bootstrapClients(host, org, repo, dir)
 	return client, err
 }
 
-// ClientFor returns a repository client for the specified repository.
+// TODO(mpherman): Deprecate in favor of ClientFromDirHost
+func (c *clientFactory) ClientFromDir(org, repo, dir string) (RepoClient, error) {
+	if c.host != "" {
+		return c.ClientFromDirHost(c.host, org, repo, dir)
+	}
+	return nil, errors.New("Client does not have a default host provided. Either set host in client or use ClientFromDirHost")
+}
+
+// TODO(mpherman): Deprecate in favor of ClientForHost
+func (c *clientFactory) ClientFor(org, repo string) (RepoClient, error) {
+	if c.host != "" {
+		return c.ClientForHost(c.host, org, repo)
+	}
+	return nil, errors.New("Client does not have a default host provided. Either set host in client or use ClientForHost")
+}
+
+// ClientForHost returns a repository client for the specified repository.
 // This function may take a long time if it is the first time cloning the repo.
 // In that case, it must do a full git mirror clone. For large repos, this can
 // take a while. Once that is done, it will do a git fetch instead of a clone,
 // which will usually take at most a few seconds.
-func (c *clientFactory) ClientFor(org, repo string) (RepoClient, error) {
+func (c *clientFactory) ClientForHost(host, org, repo string) (RepoClient, error) {
+	if c.host != "" && host != c.host {
+		return nil, errors.New("Client does not have a default host provided. Either set host in client or use ClientForHost")
```

Comment:
```suggestion
		return nil, errors.New("client does not have a default host provided. Either set host in client or use ClientForHost")
```
errors starts with lowercase